src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,26 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
 
 public class AsyncRequestProcessor {
     private final Executor executor;
+    private final Map<String, UserData> cache = new ConcurrentHashMap<>();
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        return CompletableFuture.supplyAsync(() -> {
+            if (cache.containsKey(userId)) {
+                return cache.get(userId);
+            }
+            UserData userData = new UserData(userId, "Details for " + userId);
+            cache.put(userId, userData);
+            return userData;
+        }, executor);
```

Comment:
The current implementation of 'processRequest' method may lead to a race condition where multiple threads could potentially create a new UserData object for the same userId if they access the method simultaneously. Consider using 'computeIfAbsent' method of the ConcurrentHashMap to ensure atomicity of the operation.